packages/fast-usdc/src/fast-usdc.start.js

```diff
@@ -0,0 +1,257 @@
+import { AmountMath } from '@agoric/ertp/src/amountMath.js';
+import { deeplyFulfilledObject, makeTracer, objectMap } from '@agoric/internal';
+import { Fail } from '@endo/errors';
+import { E } from '@endo/far';
+import { M } from '@endo/patterns';
+import { makeMarshal } from '@endo/marshal';
+import { FastUSDCTermsShape } from './type-guards.js';
+import { fromExternalConfig } from './utils/config-marshal.js';
+
+/**
+ * @import {DepositFacet} from '@agoric/ertp/src/types.js'
+ * @import {TypedPattern} from '@agoric/internal'
+ * @import {Instance, StartParams} from '@agoric/zoe/src/zoeService/utils'
+ * @import {Board} from '@agoric/vats'
+ * @import {ManifestBundleRef} from '@agoric/deploy-script-support/src/externalTypes.js'
+ * @import {BootstrapManifest} from '@agoric/vats/src/core/lib-boot.js'
+ * @import {LegibleCapData} from './utils/config-marshal.js'
+ * @import {FastUsdcSF, FastUsdcTerms} from './fast-usdc.contract.js'
+ */
+
+const trace = makeTracer('FUSD-Start', true);
+
+const contractName = 'fastUSDC';
+
+/**
+ * @typedef {{
+ *   terms: FastUsdcTerms;
+ *   oracles: Record<string, string>;
+ * }} FastUSDCConfig
+ */
+/** @type {TypedPattern<FastUSDCConfig>} */
+export const FastUSDCConfigShape = M.splitRecord({
+  terms: FastUSDCTermsShape,
+  oracles: M.recordOf(M.string(), M.string()),
+});
+
+/**
+ * XXX Shouldn't the bridge or board vat handle this?
```

Comment:
You already filed an issue for boardAux support :)

 - #8407

This XXX isn't about boardAux. It's about the klunky handling of storageNode+marshaller all over every contract that publishes to vstorage. `prepareRecorderKitMakers` and all that. If the bridge or board vat just handed out nodes that have methods for `.formatAndWrite(passable)` and `.makeChildNode()`, it would all be so much simpler.

Reply:
> If the bridge or board vat just handed out nodes that have methods for `.formatAndWrite(passable)` and `.makeChildNode()`, it would all be so much simpler.

Why do those nodes have to live in a separate vat?  Is it possible to build a better library function to construct such nodes?

The reason I'm asking is that right now, the storageNode and marshaller objects are decoupled to support POLA.  They're only combined in the contract itself, where the code instantiating the contract is the one to supply the privileges.  That design choice has little to do with improving the abstraction that combines them.

Reply:
> Why do those nodes have to live in a separate vat?

I'm not sure they do. The XXX is a bit of thinking-out-loud.

I'd like to avoid having contract vats make an additional durable object for each node.

And to reduce round-trips... as in #6646 , though that PR suggests round trips can be avoided in other ways.

> right now, the storageNode and marshaller objects are decoupled to support POLA.

Right... the exercise here is to explore adding "write to vstorage" power to the board vat or let the bridge vat ask the bridge for slots. The latter appeals to me.